src/ahp/mod.rs

```diff
@@ -89,7 +85,7 @@ impl<F: PrimeField> AHPForR1CS<F> {
             3 * domain_h_size + 2 * zk_bound - 3, //  mask_poly
             domain_h_size,
             domain_h_size,
-            3 * domain_k_size - 3,
+            domain_k_size - 1,
```

Comment:
`domain_k` is actually 3 times what it previously was, so this isn't actually a big saving =P

Reply:
Note: if we are in the recursion-friendly mode, the vanishing polynomial of domain K is committed, in order to efficiently prove its evaluation on point \gamma to the verifier. This vanishing polynomial has degree `domain_k_size` and would bump this one up by one.

Reply:
TODO: address this when porting marlin constraints